module/zfs/dmu_recv.c

```diff
@@ -2486,11 +2486,15 @@ dmu_recv_stream(dmu_recv_cookie_t *drc, vnode_t *vp, offset_t *voffp,
 		 * the keynvl away until then.
 		 */
 		err = dsl_crypto_recv_raw(spa_name(ra->os->os_spa),
-		    drc->drc_ds->ds_object, drc->drc_drrb->drr_type,
-		    keynvl, drc->drc_newfs);
+		    drc->drc_ds->ds_object, drc->drc_fromsnapobj,
+		    drc->drc_drrb->drr_type, keynvl, drc->drc_newfs);
 		if (err != 0)
 			goto out;
 
+		/* see comment in dmu_recv_end_sync() */
+		(void) nvlist_lookup_uint64(keynvl, "to_ivset_guid",
+		    &drc->drc_ivset_guid);
```

Comment:
This needs some error handling to make sure `&drc->drc_ivset_guid` is set consistently to zero for ENOENT and error out for any other unexpected errors

Reply:
I did this after looking with several other places in the code that do the same thing (cast to void to convert an ENOENT to a default value of 0). Is this something we want to change throughout the code?

Reply:
Depending on the situation I think this little trick can make sense.  I just don't think it's a good idea here because it assumes the passed `drc->drc_ivset_guid` value is `0`.  That happens to be the case since `dmu_recv_begin()` zeros the entire structure in `zfs_ioc_recv_impl()`, but it would be safer not to depend on that.  Rather then checking the return value you could explicitly set `drc->drc_ivset_guid = 0` before the lookup.

Reply:
fixed